### DISCUSSION TITLE
What is the workflow to create a Vertex Animation for cables

### DISCUSSION THREAD

**Discussion by wirmachenbunt** [cables-gl/cables_docs#260] [Answers & Questions] [answered]
there is this demo https://cables.gl/edit/2gX261
showcasing a vertex animation. i would like to learn how this is done in Blender or Cinema4D. If it isn't bones and not PLA...what is this ? i'm curious.

![image](https://user-images.githubusercontent.com/1138673/171666389-87787799-4226-495a-8c53-52b4966b5040.png)

**Comment by tastez** [comment, marked as answer]
I've only done this in Blender, where it is called Shape Keys, but you basically make keyframes of your vertices on a model and then fade between them.

think i followed this https://www.youtube.com/watch?v=e675Ulc7QL4

**Reply by wirmachenbunt** [reply]
cool cool, i was looking into that as well. ok, i need to get my modifier animation somehow baked into shape keys. 
cheers man !

**Reply by wirmachenbunt** [reply]
got it working 
https://cables.gl/p/VsLPA4
there is still lots to learn in blender

**Reply by akerb00m** [reply]
Hi there : )
I have been trying to get Blender shapekeys to work in Cables for a while now but i can't get it to work... feels like i have tried every possible combination of GLB export settings but i must be missing something. could you elaborate what export settings to use ? also, should i keyframe the shapekey? bake it? make a nla track? if yes turn it into an action ? ... so many possibilities  : o
Cheers !

**Reply by wirmachenbunt** [reply]
hey, i followed the youtube tutorial, is this not working for you. i can't remember setting anything special in blender. as far as i remember, if you see the shapekey animation in blender, it works in cables too. are you sure you have a timelined (like in the yt tut) shapekey animation ?

**Reply by dudumaluf** [reply]
https://cables.gl/p/611166992727b6137df6bf80?s=7e6nh4wXqboewICj
In cinemad4d Ive pretty much just baked the physics animation first internally then Ive baked it as PLA ( Point Level Animation )
its been a while, will try again and let you know the c4d method in case someone needs it,.

**Comment by n4hui** [comment]
![1](https://user-images.githubusercontent.com/65823796/202359635-b5541c47-ec75-4dc7-a83e-99fd396a1d76.jpg)
![2](https://user-images.githubusercontent.com/65823796/202359637-dd8f6368-370a-4860-bae8-1eb69a411482.jpg)
![3](https://user-images.githubusercontent.com/65823796/202359638-9ae142c3-88e9-474c-918a-2aaeba60a1cd.jpg)
![4](https://user-images.githubusercontent.com/65823796/202359640-96f9b045-99c9-46fd-88cd-1fd0755111aa.jpg)
![5](https://user-images.githubusercontent.com/65823796/202359643-1a3bc303-6679-4d5c-8f5b-d098202978fc.jpg)
![6](https://user-images.githubusercontent.com/65823796/202359629-72814858-d408-49a7-9b7d-daa6e9e3ee35.jpg)
![7](https://user-images.githubusercontent.com/65823796/202359631-efda25b0-f0bf-4311-91f0-924a58a9fada.jpg)
![8](https://user-images.githubusercontent.com/65823796/202359633-5a14be6d-8143-4489-a176-ebfaf28e1a94.jpg)

Hi! well i share you my method you need to activate the animateall and newtek mdd addons in blender. then you make a geometry and add an animate all keyframe, then advance in time and add another keyframe changing the point positions, then select the object and export the mdd format, and with the object selected import the mdd file so it bakes the keyframes for the vertex animation.

**Reply by wirmachenbunt** [reply]
[shapekeyCablesExample.zip](https://github.com/cables-gl/cables_docs/files/10029615/shapekeyCablesExample.zip)

and here is a blender project file and the cables patch using this blender output
https://cables.gl/p/VsLPA4

before i export, i do select all objects i want to export in the editor. that should be it.

**Comment by akerb00m** [comment]
Thank you everyone ! 
I really appreciate you sharing all this information !
Maybe the thing is that i have been trying to use shapekeys as they are set up in the 'object data properties panel' and have been avoiding  the 'animall' addon because it caused some pretty bad crashes in the past ; )
I can't wait to try to get shapekeys into cables again with all this new information. I have to wait until my PC is back from getting repaired but ill you posted on my progress when the time comes : )
Cheers !

**Comment by akerb00m** [comment]
@n4hui it worked !
thank you for the description with the screenshots, i really appreciate the effort !
